src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/ExternalAccountManager.java

```diff
@@ -0,0 +1,143 @@
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.cam.cl.dtg.segue.dao.SegueDatabaseException;
+import uk.ac.cam.cl.dtg.segue.dao.users.IExternalAccountDataManager;
+import uk.ac.cam.cl.dtg.segue.dos.users.EmailVerificationStatus;
+import uk.ac.cam.cl.dtg.segue.dos.users.UserExternalAccountChanges;
+import uk.ac.cam.cl.dtg.util.email.MailJetApiClientWrapper;
+import uk.ac.cam.cl.dtg.util.email.MailJetSubscriptionAction;
+
+import java.util.List;
+
+public class ExternalAccountManager {
+    private static final Logger log = LoggerFactory.getLogger(ExternalAccountManager.class);
+
+    private final IExternalAccountDataManager database;
+    private final MailJetApiClientWrapper mailjetApi;
+
+    @Inject
+    public ExternalAccountManager(final MailJetApiClientWrapper mailjetApi, final IExternalAccountDataManager database) {
+        this.database = database;
+        this.mailjetApi = mailjetApi;
+    }
+
+    public void synchroniseChangedUsers() throws ExternalAccountSynchronisationException {
+        try {
+            List<UserExternalAccountChanges> userRecordsToUpdate = database.getRecentlyChangedRecords();
+
+            for (UserExternalAccountChanges userRecord : userRecordsToUpdate) {
+
+                Long userId = userRecord.getUserId();
+                log.debug(String.format("Processing user: %s", userId));
+                try {
+
+                    String accountEmail = userRecord.getAccountEmail();
+                    String mailjetId = userRecord.getProviderUserId();
+                    JSONObject mailjetDetails;
+
+                    if (null != mailjetId) {
+                        // If there is a "mailjet_id", get the account from MailJet.
+                        mailjetDetails = mailjetApi.getAccountByIdOrEmail(mailjetId);
+                        if (userRecord.isDeleted()) {
+                            // Case: deleted from Isaac but still on MailJet:
+                            //    Expect: "deleted" but non-null "mailjet_id"
+                            //    Action: GDPR deletion, null out MailJet ID?, update provider_last_updated
+                            log.debug("Case: deletion.");
+                            deleteUserFromMailJet(mailjetId, userRecord);
+                        } else if (EmailVerificationStatus.DELIVERY_FAILED.equals(userRecord.getEmailVerificationStatus())) {
+                            // Case: DELIVERY_FAILED but already on MailJet
+                            //    Expect: DELIVERY_FAILED, but non-null "mailjet_id"
+                            //    Action: same as deletion? Or just remove from lists for now?
+                            log.debug("Case: delivery failed.");
+                            mailjetApi.updateUserSubscriptions(mailjetId, MailJetSubscriptionAction.REMOVE, MailJetSubscriptionAction.REMOVE);
+                        } else if (!accountEmail.equals(mailjetDetails.getString("Email"))) {
```

Comment:
I'd discussed this with the others. Since between 1/3 and 1/2 of users do not verify their email addresses, we're including the not verified users too (with a note that they are not verified to allow segmenting). It's worth noting that to have _changed_ an account email does require verifying the new address so this particular code path shouldn't need to worry about that either way.